README.adoc

```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
+
 ==== Prerequisites if running locally on minishift
 If you are running this tests locally on minishift, make sure that you have exposed minishift's docker-env, so that deployment can use locally built image. You can expose it by running following command.
 `eval $(minishift docker-env)`
 
 NOTE: This is not required for openshift-ci environment
 
 ==== Running End-to-End Tests
-The `make test-e2e` will take care of creating a namespace/project with a random name (or see below for enforcing the namespace). It will also create all required CRDs, role and role binding for the service account, build the Docker image for the operator and push it to the OpenShift container registry. Finally, it will deploy the operator and run the tests using
-the operator-sdk.
 
+Although the e2e tests are in the separated repository, it's still possible to run them from this repo (member-operator) and also against the current code that is at HEAD.
+There are two Makefile targets that will execute the e2e tests:
+
+* `make test-e2e` - this target clones latest changes from https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] and runs e2e tests for both operators from the master. As deployment for `member-operator` it uses the current code that is at HEAD.
+* `make test-e2e-local` - this target doesn't clone anything, but it runs run e2e tests for both operators from the directory `../toolchain-e2e`. As deployment for `member-operator` it uses the current code that is at HEAD.
+
+The tests executed within https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo will take care of creating all needed namespaces with random names (or see below for enforcing some specific namespace names).
+It will also create all required CRDs, role and role bindings for the service accounts, build the Docker images for both operators and push them to the OpenShift container registry. Finally, it will deploy the operators and run the tests using the operator-sdk.
+
+ NOTE: you can override the default namespace names where the end-to-end tests are going to be executed - eg.: `make test-e2e HOST_NS=my-host MEMBER_NS=my-member` file.
+
+===== What to do
+
+If you are still confused by the e2e location, execution and branch pairing, see the following cases and needed steps:
+
+* *Working locally:*
+** *Need to test your code using the latest version of e2e tests from https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo:*
+*** execute `make test-e2e`
+** *Need to test your code using e2e tests located in `../toolchain-e2e` repo:*
+*** `make test-e2e-local`
 
- NOTE: you can override the default namepace names where the end-to-end tests are going to be executed - eg.: `make test-e2e HOST_NS=my-host MEMBER_NS=my-member` file.
+* *Creating a PR:*
+** *Your PR doesn't need any changes in https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo:*
```

Comment:
I'm not sure why you mix underordered (`*`, `**`, etc.) and ordered (`1.`, `2.` etc) prefixes here

Reply:
it looked better from my point of view - see the host readme https://github.com/codeready-toolchain/host-operator#what-to-do
@xcoulon Feel free to change it anyhow you want and push it to my branch ;-)